### PR TITLE
(WIP) docs(coro): reorganize coro docs and add examples for new features

### PR DESCRIPTION
**Pull request by Mishura4**
- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

### CODE REVIEW

**Comment by Mishura4** [comment]
~~TODO : ~~ TODO: fix colons

**Comment by Mishura4** [comment]
👍 squash it, ship it

File: docpages/example_programs/using_coroutines/coro_simple_commands.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+\note The next example assumes you are already familiar with how to use \ref firstbot "slash commands", \ref slashcommands "parameters", and \ref discord-application-command-file-upload "sending files through a command".
+
+With coroutines, it becomes a lot easier to do several asynchronous requests for one task. As an example an "addemoji" command taking a file and a name as a parameter. This means downloading the emoji, submitting it to Discord, and finally replying, with some error handling along the way. Normally we would have to use callbacks and some sort of object keeping track of our state, but with coroutines, the function can simply pause and be resumed when we receive the response to our request :
+
+~~~~~~~~~~cpp
+#include <dpp/dpp.h>
+
+int main() {
+	dpp::cluster bot("token", dpp::i_default_intents | dpp::i_message_content);
```
Comment (Jaskowicz1) [inline docpages/example_programs/using_coroutines/coro_simple_commands.md:15]: intent

File: docpages/example_programs/using_coroutines/coro_simple_commands.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+Earlier we mentioned two other types of coroutines provided by dpp: dpp::coroutine and dpp::task. They both take their return type as a template parameter, which may be void. Both dpp::job and dpp::task start on the constructor for asynchronous execution, however only the latter can be `co_await`-ed, this allows you to retrieve its return value. If a dpp::task is destroyed before it ends, it is cancelled and will stop when it is resumed from the next `co_await`. dpp::coroutine also has a return value and can be `co_await`-ed, however it only starts when `co_await`-ing, meaning it is executed synchronously.
+
+Here is an example of a command making use of dpp::task to retrieve the avatar of a specified user, or if missing, the sender:
+
+~~~~~~~~~~cpp
+#include <dpp/dpp.h>
+
+int main() {
+	dpp::cluster bot("token", dpp::i_default_intents | dpp::i_message_content);
```
Comment (Jaskowicz1) [inline docpages/example_programs/using_coroutines/coro_simple_commands.md:90]: intent

File: docpages/example_programs/using_coroutines/expiring_buttons.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+\include{doc} coro_warn.dox
+
+In the last example we've explored how to \ref awaiting-events "await events" using coroutines, we ran into the problem of the coroutine waiting forever if the button was never clicked. Wouldn't it be nice if we could add an "or" to our algorithm, for example wait for the button to be clicked *or* for a timer to expire? I'm glad you asked! D++ offers \ref dpp::when_any "when_any" which allows exactly that. It is a templated class that can take any number of awaitable objects and can be `co_await`-ed itself, will resume when the __first__ awaitable completes and return a \ref dpp::when_any::result "result" object that allows to retrieve which awaitable completed as well as its result, in a similar way as std::variant.
+
+~~~~~~~~~~cpp
+#include <dpp/dpp.h>
+
+int main() {
+	dpp::cluster bot("token", dpp::i_default_intents | dpp::i_message_content);
```
Comment (Jaskowicz1) [inline docpages/example_programs/using_coroutines/expiring_buttons.md:11]: intent

File: docpages/include/coro_warn.dox [hunk 4]
```diff
@@ -0,0 +1 @@
+\warning D++ Coroutines are a very new feature and are currently only supported by D++ on g++ 11, clang/LLVM 14, and MSVC 19.37 or above. Additionally, D++ must be built with the CMake option DPP_CORO, and your program must both define the macro DPP_CORO and use C++20 or above. The feature is experimental and may have bugs or even crashes, please report any to <a href="https://github.com/brainboxdotcc/DPP/issues">GitHub Issues</a> or to our <a href="https://discord.gg/dpp">Discord Server</a>.
```
Comment (Jaskowicz1) [inline docpages/include/coro_warn.dox:1]: why `<a href>text</a>` and not `[text](url)`? Is `[]()` not allowed here?

Comment (Mishura4) [reply docpages/include/coro_warn.dox:1]: No this doesn't work with \include, sadly

Comment (Jaskowicz1) [reply docpages/include/coro_warn.dox:1]: Ahhh, that's silly, got ya :)

**Pull request merged**